scripts/split-e2e-tests-v2.ts

```diff
@@ -0,0 +1,371 @@
+                }
+                soloJobs.push(newSoloJob);
+                continue;
+            }
+
+            // add the test
+            currentJob.tests.push(test);
+            if(US_WEST_2){
+                currentJob.region = 'us-west-2';
+            }
+            if(USE_PARENT){
+                currentJob.useParentAccount = true;
+            }
+        
+            // create a new job once the current job is full;
+            if(currentJob.tests.length >= MAX_WORKERS){
+                osJobs.push(createRandomJob(os));
+            }
+        }
+        return [...osJobs, ...soloJobs];
+    }
+    const linuxJobs = generateJobsForOS('l');
+    const windowsJobs = generateJobsForOS('w');
+    
+    // create the new job configurations, which will be added to the "jobs"
+    // section of the CircleCI config file
+    let newJobConfigurations = {};
+    const generateJobConfigurations = (jobs: CandidateJob[]) => {
+        for(let j of jobs){
+            if(j.tests.length === 0){
+                continue;
+            }
+            const names = j.tests.map(tn => getOldJobNameWithoutSuffixes(tn)).join('_');
+            let jobName = `${j.os}_${names}`;
+            if(isMigration){
+                const startIndex = baseJobName.lastIndexOf('_');
+                jobName = jobName + baseJobName.substring(startIndex);
+            }
+
+            newJobConfigurations = {
+                ...newJobConfigurations,
+                [jobName]: {
+                    ...baseJob,
+                    environment: {
+                        ...(baseJob?.environment || {}),
+                        TEST_SUITE: j.tests.join('|'),
+                        CLI_REGION: j.region,
+                        ...(j.useParentAccount ? { USE_PARENT_ACCOUNT: 1 } : {})
+                    }
+                }
+            }
+            if(j.os === 'l'){
+                counts.l ++;
+            } else {
+                counts.w ++;
+            }
+        }
+    }
+    generateJobConfigurations(linuxJobs);
+    generateJobConfigurations(windowsJobs);
+
+
+    // Split jobs by region
+    const jobByRegion = Object.entries(newJobConfigurations).reduce((acc: Record<string, any>, entry: [string, any]) => {
+        const [jobName, job] = entry;
+        const region = job?.environment?.CLI_REGION;
+        const regionJobs = { ...acc[region], [jobName]: job };
+        return { ...acc, [region]: regionJobs };
+    }, {});
+
+    const workflows = { ...config.workflows };
+
+    if (workflows[workflowName]) {
+        const workflow = workflows[workflowName];
+
+        const workflowJob = workflow.jobs.find(j => {
+        if (typeof j === 'string') {
+            return j === baseJobName;
+        } else {
+            const name = Object.keys(j)[0];
+            return name === baseJobName;
+        }
+        });
+
+        if (workflowJob) {
+        Object.values(jobByRegion).forEach((regionJobs: any) => {
```

Comment:
Formatting. The formatting in the whole file is non-standard and should be reformatted.

Reply:
thanks - just ran formatting on all files